app/frontend/packs/components/checkbox_search_filter.js

```diff
@@ -0,0 +1,206 @@
+    this.checkboxesContainer = this.container.querySelector('.app-checkbox-filter__container')
+    this.checkboxesInnerContainer = this.checkboxesContainer.querySelector('.app-checkbox-filter__container-inner')
+    this.setup()
+  }
+
+  filterCheckboxes () {
+    const textValue = this.sanitizeText(this.textBox.value)
+    const allCheckboxes = this.getAllCheckboxes()
+
+    for (let i = 0; i < allCheckboxes.length; i++) {
+      const labelValue = this.sanitizeText(allCheckboxes[i].querySelector('.govuk-checkboxes__label').textContent)
+      if (labelValue.search(textValue) === -1) {
+        allCheckboxes[i].style.display = 'none'
+      } else {
+        allCheckboxes[i].style.display = 'block'
+      }
+    }
+
+    this.updateStatusBox({
+      foundCount: this.getAllVisibleCheckboxes().length,
+      checkedCount: this.getAllVisibleCheckedCheckboxes().length
+    })
+  }
+
+  getAllCheckboxes () {
+    return this.checkboxesContainer.querySelectorAll('.govuk-checkboxes__item')
+  }
+
+  getAllVisibleCheckboxes () {
+    return [].filter.call(this.getAllCheckboxes(), function (el) {
+      return el.style.display === 'block'
+    })
+  }
+
+  getAllVisibleCheckedCheckboxes () {
+    return [].filter.call(this.getAllVisibleCheckboxes(), function (el) {
+      return el.querySelector('.govuk-checkboxes__input').checked
+    })
+  }
+
+  getTextBoxHtml () {
+    const containerId = this.container.id
+    const label = document.createElement('label')
+    label.setAttribute('for', containerId + '-checkbox-filter__filter-input')
+    label.className = 'govuk-label govuk-visually-hidden'
+    label.innerHTML = this.params.textBox.label
+
+    const input = document.createElement('input')
+    input.className = 'app-checkbox-filter__filter-input govuk-input'
+    input.setAttribute('id', containerId + '-checkbox-filter__filter-input')
+    input.setAttribute('type', 'text')
+    input.setAttribute('aria-describedby', containerId + '-checkboxes-status')
+    input.setAttribute('aria-controls', containerId + '-checkboxes')
+    input.setAttribute('autocomplete', 'off')
+    input.setAttribute('spellcheck', 'false')
+
+    return [label, input]
+  }
+
+  getVisibleCheckboxes () {
+    const visibleCheckboxes = [].filter.call(this.checkboxes, (el) => this.isCheckboxInView(el))
+
+    // add an extra checkbox, if the label of the first is too long it collapses onto itself
+    visibleCheckboxes.push(this.checkboxes[visibleCheckboxes.length - 1])
+    return visibleCheckboxes
+  }
+
+  isCheckboxInView (checkbox) {
+    const initialOptionContainerHeight = this.checkboxesContainer.style.height
+    const optionListOffsetTop = this.checkboxesInnerContainer.offsetTop
+    const distanceFromTopOfContainer = checkbox.offsetTop - optionListOffsetTop
+    return distanceFromTopOfContainer < initialOptionContainerHeight
+  }
+
+  onTextBoxKeyUp (e) {
+    const ENTER_KEY = 13
+    if (e.keyCode === ENTER_KEY) {
+      e.preventDefault()
```

Comment:
What's the intention of this line?

Reply:
I assume this is to prevent submission of the form on return, however, I am not sure as most of this code was inherited from the prototype.